hpctestlib/microbenchmarks/mpi/osu.py

```diff
@@ -0,0 +1,163 @@
+        self.prebuild_cmds += [
+            f'cp {fullpath} {self.stagedir}',
+            f'tar xzf {tarball}',
+            f'cd {self.build_prefix}'
+        ]
+        self.build_system.config_opts = [f'--enable-{self.build_type}']
+        self.build_system.make_opts = ['-C', 'mpi']
+        self.build_system.max_concurrency = 8
+
+    @sanity_function
+    def validate_build(self):
+        return True
+
+
+class osu_benchmark_test_base(rfm.RunOnlyRegressionTest):
+    '''Base class of OSU benchmarks runtime tests'''
+
+    #: Number of warmup iterations
+    #:
+    #: This value is passed to the excutable through the -x option.
+    #:
+    #: :type: :class:`int`
+    #: :default: ``1000``
+    num_warmup_iters = variable(int, value=1000)
+
+    #: Number of iterations
+    #:
+    #: This value is passed to the excutable through the -i option.
+    #:
+    #: :type: :class:`int`
+    #: :default: ``20000``
+    num_iters = variable(int, value=20000)
+
+    #: Maximum message size
+    #:
+    #: Both the performance and the sanity checks will be done
+    #: for this message size
+    #:
+    #: :type: :class:`int`
+    #: :default: ``8``
+    message_size = variable(int, value=8)
+
+    #: Performance message size
+    #:
+    #: This value is used for the performance checks
+    #:
+    #: :type: :class:`int`
+    #: :default: ``8``
+    perf_msg_size = variable(int, value=8)
+
+    #: Accelerator device type
+    #:
+    #: Use accelerator device buffers, i.e cuda, openacc or rocm
+    #:
+    #: :type: :class:`str`
+    #: :default: ``None``
+    device_buffers = variable(str, type(None), value=None)
+
+    executables = {
+        'collective': ['osu_alltoall', 'osu_allreduce'],
+        'pt2pt': ['osu_bw', 'osu_latency']
+    }
+    osu_binaries = fixture(build_osu_benchmarks, scope='environment')
+
+    @run_after('setup')
+    def set_executable(self):
+        if self.executable in self.executables['collective']:
+            benchmark_type = 'collective'
+        elif self.executable in self.executables['pt2pt']:
+            benchmark_type = 'pt2pt'
+
+        self.executable = os.path.join(
+            self.osu_binaries.stagedir, self.osu_binaries.build_prefix,
+            'mpi', benchmark_type, self.executable
+        )
+        max_message_size = max(self.message_size, self.message_size)
+        self.executable_opts = ['-m', f'{max_message_size}',
+                                '-x', f'{self.num_warmup_iters}',
+                                '-i', f'{self.num_iters}']
+        if self.device_buffers:
+            self.executable_opts += ['-d', f'{self.device_buffers}']
+
+        if benchmark_type == 'pt2pt':
+            self.executable_opts += ['D', 'D']
+
+    @sanity_function
+    def validate_test(self):
+        return sn.assert_found(rf'^{self.message_size}', self.stdout)
+
+
+@rfm.simple_test
+class osu_bandwidth(osu_benchmark_test_base):
```

Comment:
```suggestion
class osu_bandwidth_check(osu_benchmark_test_base):
```

Same

Reply:
This is not currently a "final" test because it's not complete. It's missing the `executable`. The idea is that anybody can run the tests from the library directly from the command line.